AEPAssurance/Source/AssuranceEventChunker.swift

```diff
@@ -102,4 +110,46 @@ struct AssuranceEventChunker {
         }
         return chunkedEvents
     }
+    
+    ///
+    /// Stitches chunked events together into one `AssuranceEvent` where the stitched events chunkData is now the new event's payload
+    ///
+    /// - Parameter chunkedEvents: An array of chunked AssuranceEvents
+    /// - Returns: An `AssuranceEvent` which has the stitched data as the payload
+    func stitch(_ chunkedEvents: [AssuranceEvent]) -> AssuranceEvent? {
+        //exit early if chunkedEvents is empty
+        guard !chunkedEvents.isEmpty else { return nil }
+        
+        var stitchedString = ""
+        // Stitch chunked payload data together
+        for event in chunkedEvents {
+            // Extract the payload string and unescape it. Currently escaped by the services
+            guard let payloadString = event.payload?[AssuranceConstants.AssuranceEvent.PayloadKey.CHUNK_DATA]?.stringValue else {
+                Log.warning(label: AssuranceConstants.LOG_TAG, "Error while attempting to stitch chunked event: Chunk payload was not in proper string format.")
+                return nil
+            }
+            
+            stitchedString += payloadString
+        }
+        guard let stitchedStringData = stitchedString.data(using: .utf8) else {
+            Log.warning(label: AssuranceConstants.LOG_TAG, "Error while attempting to create data from stitched string")
+            return nil
+        }
+        
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .millisecondsSince1970
+        var decodedPayload: [String: AnyCodable]? = nil
+        do {
+            decodedPayload = try decoder.decode([String:AnyCodable].self, from: stitchedStringData)
+        } catch {
+            Log.warning(label: AssuranceConstants.LOG_TAG, "Error while attempting to decode stitched JSON data: \(error)")
+            return nil
+        }
+        let referenceEvent = chunkedEvents[0]
+        return AssuranceEvent(type: referenceEvent.type,
+                              payload: decodedPayload,
+                              timestamp: referenceEvent.timestamp ?? Date(),
+                              vendor: referenceEvent.vendor)
```

Comment:
Okay. I see it here. In Android I just used the default vendor. I will update it.